src/idl/include/idl/tree.h

```diff
@@ -341,6 +346,14 @@ struct idl_struct {
   idl_extensibility_t extensibility;
 };
 
+typedef struct idl_forward_decl idl_forward_decl_t;
+struct idl_forward_decl {
+  idl_node_t node;
+  struct idl_name *name;
+  struct idl_scoped_name *scoped_name;
```

Comment:
Scoped name is already stored with the declaration itself. Is there a reason it's repeated here?

Reply:
No there is no reason, I just missed that, fixed in c80f50d21d8712247a7a6c7c48aae7823ce6dc0a